dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java

```diff
@@ -189,7 +187,10 @@ public DefaultTrackerBundleService( TrackerPreheatService trackerPreheatService,
     {
         TrackerBundle trackerBundle = params.toTrackerBundle();
         TrackerPreheatParams preheatParams = params.toTrackerPreheatParams();
-        preheatParams.setUser( getUser( preheatParams.getUser(), preheatParams.getUserId() ) );
+        if ( preheatParams.getUser() == null )
```

Comment:
Do we really need to set the user here?
It should be already be set before and just passed during the conversions

Reply:
Yes, it may not be needed. I didn't want to change too much of the original logic, I'll try to remove it and see what happens.

Reply:
From a security perspective I would personally rather like us to cast an exception here.
This is what I'm trying to move towards, it's much better to break fast and tell the user it's a bug and that he/she should report it.

Reply:
@netroms yes, I agree. The problem for me is that it is a bit unclear ATM how the system (Tracker) should behave in regards to users. My understanding is that Tracker can import data associated to a User which is not the currently logged in User (@mortenoh can comment on this). If this is no longer the case, I think the architecture could be simplified a bit, because we can assume that the User 1) must be always present 2) we can fetch the user from the security context rather than picking it from the database. Thoughts?

Reply:
This logic shouldnt trigger _currently_ since there is no way the system can start a tracker importer job itself (It has to be initiated by a user now). But we need a quick discussion to figure out how this will work with synchronization.

Reply:
@luciano-fiandesio yes you are right, but this also applies to the current async logic, the threaded user context is lost when we thread off (we do have some threading utils for doing threads with security context though), and it should be possible to do imports on behalf of other users.. if this is really something we need is debatable (the general idea is that the superuser can import using a lowered level of authority)